networksurvey/src/main/java/com/craxiom/networksurvey/fragments/CodeScannerFragment.java

```diff
@@ -0,0 +1,98 @@
+package com.craxiom.networksurvey.fragments;
+
+import android.app.Activity;
+import android.content.SharedPreferences;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+import androidx.navigation.Navigation;
+
+import com.budiyev.android.codescanner.CodeScanner;
+import com.budiyev.android.codescanner.CodeScannerView;
+import com.craxiom.networksurvey.R;
+import com.craxiom.networksurvey.fragments.model.ScannedSettings;
+import com.google.gson.Gson;
+
+/**
+ * Fragment responsible for QR code scanning. Leverages an open source code scanning library from
+ * Yuriy Budiev.
+ *
+ * @since 1.7.0
+ */
+public class CodeScannerFragment extends Fragment
+{
+    private CodeScanner codeScanner;
+
+    @Nullable
+    @Override
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
+                             @Nullable Bundle savedInstanceState)
+    {
+        final Activity activity = getActivity();
+        View root = inflater.inflate(R.layout.fragment_scanner, container, false);
+
+        CodeScannerView scannerView = root.findViewById(R.id.scanner_view);
+        codeScanner = new CodeScanner(activity, scannerView);
+        codeScanner.setDecodeCallback(result -> activity.runOnUiThread(() -> {
+
+            if (!result.getText().isEmpty())
+            {
+                SharedPreferences preferences = android.preference.PreferenceManager.getDefaultSharedPreferences(getContext());
+                SharedPreferences.Editor edit = preferences.edit();
+                edit.putBoolean("mqtt_mdm_override", true);
+                final ScannedSettings scannedSettings = new Gson().fromJson(result.getText(), ScannedSettings.class);
+
+                if (scannedSettings.getHost() != null)
+                {
+                    edit.putString("mqtt_connection_host", scannedSettings.getHost());
```

Comment:
There are constants that you can use for these. E.g. `com.craxiom.mqttlibrary.MqttConstants#PROPERTY_MQTT_CONNECTION_HOST`